video/out/win32/displayconfig.c

```diff
@@ -138,3 +166,89 @@ double mp_w32_displayconfig_get_refresh_rate(const wchar_t *device)
     talloc_free(ctx);
     return freq;
 }
+
+wchar_t* mp_w32_displayconfig_get_device_from_friendly_name(
+    const wchar_t *monitor_friendly_device_name)
+{
+    void *ctx = talloc_new(NULL);
+    wchar_t* gdi_device_name = NULL;
+
+    // Get the current display configuration
+    UINT32 num_paths;
+    DISPLAYCONFIG_PATH_INFO* paths;
+    UINT32 num_modes;
+    DISPLAYCONFIG_MODE_INFO* modes;
+    if (get_config(ctx, &num_paths, &paths, &num_modes, &modes))
+        goto end;
+
+    // Get the path for the specified monitor
+    DISPLAYCONFIG_PATH_INFO* path;
+    if (!(path = get_path_from_friendly(num_paths,
+                                        paths,
+                                        monitor_friendly_device_name)))
+        goto end;
+
+    // Get the GDI device name from the path
+    DISPLAYCONFIG_SOURCE_DEVICE_NAME source = {
+        .header = {
+            .size = sizeof source,
+            .type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME,
+            .adapterId = path->sourceInfo.adapterId,
+            .id = path->sourceInfo.id,
+        }
+    };
+    if (DisplayConfigGetDeviceInfo(&source.header) != ERROR_SUCCESS)
+        goto end;
+
+    gdi_device_name = talloc_memdup(NULL,
+                                    source.viewGdiDeviceName,
+                                    sizeof(source.viewGdiDeviceName));
+
+end:
+    talloc_free(ctx);
+    return gdi_device_name;
+}
+
+wchar_t* mp_w32_displayconfig_get_friendly_name_from_device(
+    const wchar_t *device)
+{
+    void *ctx = talloc_new(NULL);
+    wchar_t* monitor_friendly_device_name = NULL;
+
+    // Get the current display configuration
+    UINT32 num_paths;
+    DISPLAYCONFIG_PATH_INFO* paths;
+    UINT32 num_modes;
+    DISPLAYCONFIG_MODE_INFO* modes;
+    if (get_config(ctx, &num_paths, &paths, &num_modes, &modes))
+        goto end;
+
+    // Get the path for the specified monitor
+    DISPLAYCONFIG_PATH_INFO* path;
+    if (!(path = get_path(num_paths, paths, device)))
+        goto end;
+
+    // Get the friendly name from the path
+    DISPLAYCONFIG_TARGET_DEVICE_NAME target = {
+        .header = {
+            .size = sizeof target,
+            .type = DISPLAYCONFIG_DEVICE_INFO_GET_TARGET_NAME,
+            .adapterId = path->targetInfo.adapterId,
+            .id = path->targetInfo.id,
+        }
+    };
+    if (DisplayConfigGetDeviceInfo(&target.header) != ERROR_SUCCESS)
+        goto end;
+
+    if (!target.flags.friendlyNameFromEdid)
+        goto end;
+
+    monitor_friendly_device_name = talloc_memdup(
+        NULL,
+        target.monitorFriendlyDeviceName,
+        sizeof(target.monitorFriendlyDeviceName));
+
+end:
+    talloc_free(ctx);
+    return monitor_friendly_device_name;
+}
```

Comment:
Could you move common part to separate function, just to avoid duplication, even if it's not that much.

Reply:
You could move lines 230-248 and 190-204 to separate functions and call in common function based on flag which one name you want to get.